src/main/java/net/minestom/server/item/ItemMetaBuilder.java

```diff
@@ -0,0 +1,155 @@
+package net.minestom.server.item;
+
+import net.kyori.adventure.text.Component;
+import net.minestom.server.item.attribute.ItemAttribute;
+import net.minestom.server.utils.NBTUtils;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jglrxavpok.hephaistos.nbt.NBTCompound;
+
+import java.util.*;
+import java.util.function.Supplier;
+
+public abstract class ItemMetaBuilder implements Cloneable {
+
+    protected int damage;
+    protected boolean unbreakable;
+    protected int hideFlag;
+    protected Component displayName;
+    protected List<Component> lore = new ArrayList<>();
+    protected Map<Enchantment, Short> enchantmentMap = new HashMap<>();
+    protected List<ItemAttribute> attributes = new ArrayList<>();
+    protected int customModelData;
+
+    protected NBTCompound originalNBT;
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder damage(int damage) {
+        this.damage = damage;
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder unbreakable(boolean unbreakable) {
+        this.unbreakable = unbreakable;
+        return this;
+    }
+
+    @Contract("_ -> this")
+    public @NotNull ItemMetaBuilder hideFlag(int hideFlag) {
```

Comment:
I think you could remove this - the enum is fine.

Reply:
Gues I would have to make the other hideFlag method Nullable, though this method is convenient when reading from vanilla nbt